components/nimbus/src/stateful/client/http_client.rs

```diff
@@ -13,18 +13,32 @@
 //!
 //! But the simple subset implemented here meets our needs for now.
 
+use std::sync::Arc;
+
 use crate::error::Result;
 use crate::schema::parse_experiments;
 use crate::stateful::client::{Experiment, SettingsClient};
-use remote_settings::RemoteSettings;
+use crate::NimbusError;
+use remote_settings::{RemoteSettingsClient, RemoteSettingsError};
+use serde_json::json;
 
-impl SettingsClient for RemoteSettings {
+impl SettingsClient for Arc<RemoteSettingsClient> {
     fn get_experiments_metadata(&self) -> Result<String> {
         unimplemented!();
     }
 
     fn fetch_experiments(&self) -> Result<Vec<Experiment>> {
-        let resp = self.get_records_raw()?;
-        parse_experiments(&resp.text())
+        let records = self.get_records(false).ok_or(RemoteSettingsError::Other {
+            reason: "Unable to fetch experiment records".to_owned(),
+        })?;
+        let data_vec: Vec<_> = records
+            .into_iter()
+            .map(|r| r.fields.get("data").cloned())
+            .collect();
+        let wrapped_data = json!({ "data": data_vec });
+        let resp = serde_json::to_string(&wrapped_data).map_err(|e| {
+            NimbusError::JSONError("resp = serde_json::to_string".to_owned(), e.to_string())
```

Comment:
This should mention that its in SettingsClient::fetch_experiments so we can guarantee its a unique string